File: frontend/src/microservices/hotel/objects/RoomTypeSummaryResponse.ts
// 本文件定义 hotel 模块的 `RoomTypeSummaryResponse`，作为摘要响应数据并提供 JSON 编解码。

export type RoomTypeSummaryResponse = {
  roomTypeId: string
  roomTypeName: string
  capacity: number
  bedType: string
  basePrice: string
  currency: string
  imageUrl: string | null
  status: string
  isBookableForRequestedStay: boolean
  availableRoomsForRequestedStay: number | null
}
export const roomTypeSummaryResponseFromJson = (json: string): RoomTypeSummaryResponse =>
  JSON.parse(json) as RoomTypeSummaryResponse

export const roomTypeSummaryResponseToJson = (value: RoomTypeSummaryResponse): string =>
  JSON.stringify(value)

File: backend/src/main/scala/microservices/hotel/objects/RoomTypeSummaryResponse.scala
package com.typesafe.travel.hotel.objects

import io.circe.{Decoder, Encoder}
import io.circe.generic.semiauto.{deriveDecoder, deriveEncoder}

final case class RoomTypeSummaryResponse(
    roomTypeId: String,
    roomTypeName: String,
    capacity: Int,
    bedType: String,
    basePrice: String,
    currency: String,
    imageUrl: Option[String],
    status: String,
    isBookableForRequestedStay: Boolean,
    availableRoomsForRequestedStay: Option[Int]
)

object RoomTypeSummaryResponse:
  given sourceEncoder: Encoder[RoomTypeSummaryResponse] = deriveEncoder
  given sourceDecoder: Decoder[RoomTypeSummaryResponse] = deriveDecoder
